src/main/java/site/coach_coach/coach_coach_server/routine/controller/RoutineController.java

```diff
@@ -26,35 +24,26 @@ public class RoutineController {
 
 	@GetMapping("/v1/routines")
 	public ResponseEntity getRoutineList(@AuthenticationPrincipal CustomUserDetails userDetails,
-		@RequestParam(name = "coachId", required = false) Long coachId) {
-		Long userId = userDetails.getUserId();
-		RoutineListRequest routineListRequest = new RoutineListRequest(userId, coachId);
-
-		if (coachId == null) {
-			return getRoutineListByMyself(routineListRequest);
+		@RequestParam(name = "coachId", required = false) Long coachIdParam,
+		@RequestParam(name = "userId", required = false) Long userIdParam) {
+		Long userIdByJwt = userDetails.getUserId();
+		RoutineListRequest routineListRequest;
+
+		if (coachIdParam == null && userIdParam == null) {
+			routineListRequest = new RoutineListRequest(userIdByJwt, null);
+			List<RoutineForListDto> routineListByMyself = routineService.getRoutineForList(routineListRequest);
+			return ResponseEntity.ok(routineListByMyself);
+		} else if (coachIdParam == null) {
+			Long coachId = routineService.getCoachId(userIdByJwt);
```

Comment:
```suggestion
		@RequestParam(name = "coachId", required = false) Long coachIdParam,
		@RequestParam(name = "userId", required = false) Long userIdParam) {
		Long userIdByJwt = userDetails.getUserId();
		RoutineListRequest routineListRequest = createRoutineListRequest(userIdByJwt, coachIdParam, userIdParam);

    routineService.getIsMatching(routineListRequest);

    List<RoutineForListDto> routineListByCoach = routineService.getRoutineForList(routineListRequest);
    return ResponseEntity.ok(routineListByCoach);
```

이런 식으로 수정을 하고 아래에 새 메서드를 만드는 건 어떨까요?
```java
private RoutineListRequest createRoutineListRequest(Long userIdByJwt, Long coachIdParam, Long userIdParam) {
    if (coachIdParam == null && userIdParam == null) {
        return new RoutineListRequest(userIdByJwt, null);
    } else if (coachIdParam == null) {
        Long coachId = routineService.getCoachId(userIdByJwt);
        return new RoutineListRequest(userIdParam, coachId);
    } else {
        return new RoutineListRequest(userIdByJwt, coachIdParam);
    }
}
```

Reply:
오우 코드가 깔끔해지고 좋네요! 
한번 분리해 보겠습니다.

Reply:
그은데 사실 갑자기 생각해보니,, 컨트롤러에서 이걸 다 해야되는지 모르겟네오 서비스딴에서 비즈니스 로직을 다 해야될텐데..흠.....ㅠㅠ